src/evaluators/gather/PHAL_GatherScalarNodalParameter_Def.hpp

```diff
@@ -361,10 +361,13 @@ evaluateFields(typename Traits::EvalData workset)
   bool g_xp_is_active = !workset.hessianWorkset.hess_vec_prod_g_xp.is_null();
   bool g_px_is_active = !workset.hessianWorkset.hess_vec_prod_g_px.is_null();
   bool g_pp_is_active = !workset.hessianWorkset.hess_vec_prod_g_pp.is_null();
+  bool f_xp_is_active = !workset.hessianWorkset.hess_vec_prod_f_xp.is_null();
+  bool f_px_is_active = !workset.hessianWorkset.hess_vec_prod_f_px.is_null();
+  bool f_pp_is_active = !workset.hessianWorkset.hess_vec_prod_f_pp.is_null();
 
   // Are we differentiating w.r.t. this parameter?
-  const bool is_active = (workset.dist_param_deriv_name == this->param_name) && (g_px_is_active||g_pp_is_active);
-  const bool is_direction_active = (workset.hessianWorkset.dist_param_deriv_direction_name == this->param_name) && (g_xp_is_active || g_pp_is_active);
+  const bool is_active = (workset.dist_param_deriv_name == this->param_name) && (g_px_is_active||g_pp_is_active||f_px_is_active||f_pp_is_active);
```

Comment:
please add a comment briefly explaining what `is_active` and `is_direction_active` are